src/Agent.Plugins/GitCliManager.cs

```diff
@@ -77,27 +82,53 @@ public bool EnsureGitLFSVersion(Version requiredVersion, bool throwOnNotMatch)
             return gitLfsVersion >= requiredVersion;
         }
 
+        public Tuple<string, string> GetInternalGitPaths(AgentTaskPluginExecutionContext context, bool useLatestGitVersion)
+        {
+            string agentHomeDir = context.Variables.GetValueOrDefault("agent.homedirectory")?.Value;
+            ArgUtil.NotNullOrEmpty(agentHomeDir, nameof(agentHomeDir));
+
+            string gitPath;
+
+            if (useLatestGitVersion)
+            {
+                gitPath = Path.Combine(agentHomeDir, "externals", "ff_git", "cmd", $"git.exe");
+            }
+            else
+            {
+                gitPath = Path.Combine(agentHomeDir, "externals", "git", "cmd", $"git.exe");
+            }
+
+            context.Debug($@"The useLatestGitVersion property is set to ""{useLatestGitVersion}"" so the Git path is ""{gitPath}""");
+
+            string gitLfsPath;
+
+            if (PlatformUtil.BuiltOnX86)
+            {
+                gitLfsPath = Path.Combine(agentHomeDir, "externals", "git", "mingw32", "bin", "git-lfs.exe");
+            }
+            else
+            {
+                gitLfsPath = Path.Combine(agentHomeDir, "externals", "git", "mingw64", "bin", "git-lfs.exe");
```

Comment:
is git-lfs version changing?  if so, would be good to gate on useLatestGitVersion

Reply:
The Git-Lfs version does not change, this is placed in a separate method to resolve paths in one place.